Funcky/Extensions/EnumerableExtensions/MinOrNone.cs

```diff
@@ -0,0 +1,337 @@
+        public static Option<TSource> MinOrNone<TSource>(this IEnumerable<Option<TSource>> source, RequireClass<TSource>? ω = null)
+            where TSource : class
+            => source.WhereSelect(Identity).MinOrNone(Identity);
+
+        /// <summary>
+        /// Returns the minimum value in a sequence of optional generic values compared by a Comparer{T}. If the sequence only consists of none or is empty it returns None.
+        /// </summary>
+        /// <param name="source">A sequence of optional generic values of TSource to determine the minimum value of.</param>
+        /// <param name="ω">Dummy Parameter, do not use.</param>
+        /// <returns>The minimum value in the sequence or None.</returns>
+        [Pure]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0027:Public API with optional parameter(s) should have the most parameters amongst its public overloads.", Justification = "The RequireStruct type is only for type deduction of the right overload.")]
+        public static Option<TSource> MinOrNone<TSource>(this IEnumerable<Option<TSource>> source, RequireStruct<TSource>? ω = null)
+            where TSource : struct
+            => source.WhereSelect(Identity).MinOrNone(Identity);
+
+        /// <summary>
+        ///  Invokes a transform function on each element of a sequence and returns the minimum from the generic values compared by a Comparer{T}. If the transforemd sequence only consists of none or is empty it returns None.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the elements of source.</typeparam>
+        /// <param name="source">A sequence of generic values of TSource to determine the minimum value of.</param>
+        /// <typeparam name="TResult">The type of the value returned by selector.</typeparam>
+        /// <param name="selector">A transform function to apply to each element.</param>
+        /// <param name="ω">Dummy Parameter, do not use.</param>
+        /// <returns>The minimum value in the sequence or None.</returns>
+        [Pure]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0027:Public API with optional parameter(s) should have the most parameters amongst its public overloads.", Justification = "The RequireClass type is only for type deduction of the right overload.")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "The RequireClass type is needed on each overload.")]
+        public static Option<TResult> MinOrNone<TSource, TResult>(this IEnumerable<TSource> source, Func<TSource, TResult> selector, RequireClass<TResult>? ω = null)
+            where TResult : class
+            => source.Aggregate(Option<TResult>.None(), (min, current) => Option.Some(min.Match(selector(current), m => Comparer<TResult>.Default.Compare(m, selector(current)) < 0 ? selector(current) : m)));
```

Comment:
`Comparer<TResult>.Default.Compare(m, selector(current)) < 0 ? selector(current) : m)` could be extracted into a generic `Min` function.

Reply:
Adressed in d194b57c